clap_complete/tests/testsuite/engine.rs

```diff
@@ -981,6 +981,167 @@ a_pos,c_pos"
     );
 }
 
+#[test]
+fn suggest_allow_hyhpen() {
+    let mut cmd = Command::new("exhaustive")
+        .arg(
+            clap::Arg::new("format")
+                .long("format")
+                .short('F')
+                .allow_hyphen_values(true)
+                .value_parser(["--json", "--toml", "--yaml"]),
+        )
+        .arg(clap::Arg::new("json").long("json"));
+
+    assert_data_eq!(complete!(cmd, "--format --j[TAB]"), snapbox::str!["--json"]);
+    assert_data_eq!(complete!(cmd, "-F --j[TAB]"), snapbox::str!["--json"]);
+    assert_data_eq!(complete!(cmd, "--format --t[TAB]"), snapbox::str!["--toml"]);
+    assert_data_eq!(complete!(cmd, "-F --t[TAB]"), snapbox::str!["--toml"]);
+
+    assert_data_eq!(
+        complete!(cmd, "--format --[TAB]"),
+        snapbox::str![
+            "--json
+--toml
+--yaml"
+        ]
+    );
+
+    assert_data_eq!(
+        complete!(cmd, "-F --[TAB]"),
+        snapbox::str![
+            "--json
+--toml
+--yaml"
+        ]
+    );
+
+    assert_data_eq!(
+        complete!(cmd, "--format --json --j[TAB]"),
+        snapbox::str!["--json"]
+    );
+
+    assert_data_eq!(
+        complete!(cmd, "-F --json --j[TAB]"),
+        snapbox::str!["--json"]
+    );
+}
+
+#[test]
+fn suggest_positional_long_allow_hyhpen() {
+    let mut cmd = Command::new("exhaustive")
+        .arg(
+            clap::Arg::new("format")
+                .long("format")
+                .short('F')
+                .allow_hyphen_values(true)
+                .value_parser(["--json", "--toml", "--yaml"]),
+        )
+        .arg(
+            clap::Arg::new("positional_a")
+                .value_parser(["--pos_a"])
+                .index(1)
+                .allow_hyphen_values(true),
+        )
+        .arg(
+            clap::Arg::new("positional_b")
+                .index(2)
+                .value_parser(["pos_b"]),
+        );
+
+    assert_data_eq!(
+        complete!(cmd, "--format --json --pos[TAB]"),
+        snapbox::str!["--pos_a"]
+    );
+    assert_data_eq!(
+        complete!(cmd, "-F --json --pos[TAB]"),
+        snapbox::str!["--pos_a"]
+    );
+
+    assert_data_eq!(
+        complete!(cmd, "--format --json --pos_a [TAB]"),
+        snapbox::str![
+            "--format
+--help	Print help
+-F
+-h	Print help
+pos_b"
+        ]
+    );
+    assert_data_eq!(
+        complete!(cmd, "-F --json --pos_a [TAB]"),
+        snapbox::str![
+            "--format
+--help	Print help
+-F
+-h	Print help
+pos_b"
+        ]
+    );
+
+    assert_data_eq!(
+        complete!(cmd, "--format --json --pos_a p[TAB]"),
+        snapbox::str!["pos_b"]
+    );
+    assert_data_eq!(
+        complete!(cmd, "-F --json --pos_a p[TAB]"),
+        snapbox::str!["pos_b"]
+    );
+}
+
+#[test]
+fn suggest_positional_short_allow_hyhpen() {
```

Comment:
The fact that these tests aren't changed in the follow up commit either means they are not testing the new behavior or they fail on this commit.  Please update them to pass.

See #5602 as an example of this

Reply:
I'm not sure I understand what's wrong because they fail on https://github.com/clap-rs/clap/pull/5628/commits/6c4b9302b35ce4737018562c86eb05445e2d4be5 but do pass on https://github.com/clap-rs/clap/pull/5628/commits/b1117176774b329b3effe451da567825ee58e35e

Reply:
Tests should pass on each commit.  The purpose for adding tests before a code change is to show the behavior before the code change so you can easily tell what the intended behavior change is.

Reply:
Oopsie daisy, I had the wrong idea!